### PR TITLE
CSV export from all submissions

### LINKED ISSUE #3981: Improve Submission CSV export functionality
**Is your feature request related to an existing functionality? Please describe.**

Submission CSV export.

**Describe the solution you'd like**

Permit the download application data from the All Submissions page - as filtered by the user - as a CSV as file.

Initially, restrict permissions to Staff/Staff Admin.

**Priority**

- Low priority (annoying, would be nice to not see)

**Affected roles**

- Staff

### PR DESCRIPTION
**Pull request by bickelj**
Fixes #3981

## Test Steps

- As reviewer, visit All Submissions, see no CSV export button
- As staff, visit All Submissions
- Filter the submissions by one or more columns such that some submissions are removed and some remain
- Click Export CSV
- Download and open the CSV file
- Verify that only the filtered submissions show up in the CSV
- Verify that application/submission columns with data are present

### CODE REVIEW

File: hypha/apply/funds/templates/funds/includes/table_filter_and_search.html [hunk 1]
```diff
@@ -66,6 +66,11 @@ <h2 class="text-xl mb-2">{{ heading }}</h2>
                     <a href="{% modify_query archived='1' %}">{% trans "Show archived" %}</a>
                 {% endif %}
             {% endif %}
+            {% if request.user.is_apply_staff or request.user.is_staff %}
```
Comment (frjo) [inline hypha/apply/funds/templates/funds/includes/table_filter_and_search.html]: Please reuse `can_export_submissions` permission.

File: hypha/apply/funds/views.py [hunk 2]
```diff
@@ -226,6 +224,17 @@ def get_context_data(self, **kwargs):
             **kwargs,
         )
 
+    def dispatch(self, request, *args, **kwargs):
+        disp = super().dispatch(request, *args, **kwargs)
+        if "export" in request.GET and (
+            self.request.user.is_staff or self.request.user.is_apply_staff
```
Comment (frjo) [inline hypha/apply/funds/views.py]: Reuse `can_export_submissions` here as well.

**Review by frjo — CHANGES_REQUESTED** [review]

**Review by frjo — APPROVED** [review]

File: hypha/apply/funds/templates/funds/includes/table_filter_and_search.html [hunk 3]
```diff
@@ -66,6 +66,11 @@ <h2 class="text-xl mb-2">{{ heading }}</h2>
                     <a href="{% modify_query archived='1' %}">{% trans "Show archived" %}</a>
                 {% endif %}
             {% endif %}
+            {% with request.get_full_path as path %}
+                {% if 'submission' in path and can_export %}
+                    <a href="{{path}}{% if '?' in path %}&{%else%}?{% endif %}export=true"><button type="submit" class="export-btn-width button--primary">{% trans 'Export CSV' %}</button></a>
```
Comment (frjo) [inline hypha/apply/funds/templates/funds/includes/table_filter_and_search.html]: Missing a "button" class.

Comment (frjo) [reply hypha/apply/funds/templates/funds/includes/table_filter_and_search.html]: Can we style the a-tag as a button directly?

Comment (frjo) [reply hypha/apply/funds/templates/funds/includes/table_filter_and_search.html]: Replace `export-btn-width` with `w-max` (tailwind).

**Review by frjo — CHANGES_REQUESTED** [review]

File: hypha/apply/funds/templates/funds/includes/table_filter_and_search.html (hunk 3 above)
Comment (bickelj) [reply hypha/apply/funds/templates/funds/includes/table_filter_and_search.html]: I couldn't see any difference with `max-w-full` class, but the anchor when styled as `button--primary` looks OK.

File: hypha/apply/funds/templates/funds/includes/table_filter_and_search.html [hunk 4]
```diff
@@ -66,6 +66,11 @@ <h2 class="text-xl mb-2">{{ heading }}</h2>
                     <a href="{% modify_query archived='1' %}">{% trans "Show archived" %}</a>
                 {% endif %}
             {% endif %}
+            {% with request.get_full_path as path %}
+                {% if 'submission' in path and can_export %}
+                    <a href="{{path}}{% if '?' in path %}&{%else%}?{% endif %}export=true" class="button--primary">{% trans 'Export CSV' %}</a>
```
Comment (frjo) [inline hypha/apply/funds/templates/funds/includes/table_filter_and_search.html]: It should be "button button--primary".

File: hypha/static_src/sass/components/_button.scss [hunk 5]
```diff
@@ -437,3 +437,7 @@
         }
     }
 }
+
+.export-btn-width {
+    width: max-content;
+}
```
Comment (frjo) [inline hypha/static_src/sass/components/_button.scss]: I think this can be removed now.

**Review by frjo — CHANGES_REQUESTED** [review]

File: hypha/apply/funds/templates/funds/includes/table_filter_and_search.html (hunk 4 above)
Comment (bickelj) [reply hypha/apply/funds/templates/funds/includes/table_filter_and_search.html]: So many redundant classes but OK, I did it : )

**Review by frjo — APPROVED** [review]

**Review by wes-otf — APPROVED** [review]

**Comment by wes-otf** [comment]
Works great, thanks for this @bickelj!

**Pull request merged**